### ISSUE TITLE
[DOCS] Changelog of 3006.3 inconsistent with release contents

### ISSUE THREAD

**Issue by tj90241** [saltstack/salt#65162]
In the [changelog](https://github.com/saltstack/salt/blob/v3006.3/CHANGELOG.md) the inclusion of this commit was mentioned:
* Switch to FullArgSpec since Py 3.11 no longer has ArgSpec, deprecated since Py 3.

However, the commit is not actually in the release...

If it was, we would expect to see e.g.:
```
-- a/tests/unit/states/test_module.py
+++ b/tests/unit/states/test_module.py
@@ -4,7 +4,7 @@
 
 
 import logging
-from inspect import ArgSpec
+from inspect import FullArgSpec
```

However, it's missing and we're still using ArgSpec, at least in the tests:
https://github.com/saltstack/salt/blob/v3006.3/tests/unit/states/test_module.py#L7

I noticed this because I maintain Debian bookworm packaging for Salt and it FTBs without this commit.

cc @Ch3LL , the author of the `v3006.3_docs` tag

**Comment by s0undt3ch** [comment]
This was apparently some bad `grep`'ing :grin: 

Luckily, only on the tests.

**Comment by dwoz** [comment]
This is fixed and will go into `3006.9`
